### PR TITLE
Allow specifying additional environment settings

### PR DESCRIPTION
**Pull request by sascha-andres**
Some tools need to have environment settings. This feature adds the ability to provide settings on a per task basis.

### CODE REVIEW

**Comment by andreynering** [comment]
Hmm... we could just set `vars` as env variables instead? Is there any benefit having 2 separated options?

File: task.go [hunk 1]
```diff
@@ -104,7 +106,7 @@ func RunTask(name string) error {
 	}
 
 	for i := range t.Cmds {
-		if err = t.runCommand(i); err != nil {
+		if err = t.runCommand(i, t.Env); err != nil {
```
Comment (andreynering) [inline task.go]: Don't need to pass `Env` as parameter. Task is available as a receiver of the func.

File: task.go [hunk 2]
```diff
@@ -151,6 +153,21 @@ func (t *Task) runCommand(i int) error {
 	if dir != "" {
 		cmd.Dir = dir
 	}
+	if nil != envVariables {
```
Comment (andreynering) [inline task.go]:
```go
t.Env != nil
```

**Comment by sascha-andres** [comment]
@andreynering  We use to have services running that take decide a mandator based on an environment. Devs set their mandator globally and when they wanted to test with another they had to source the profile. The ability to set the env of the child process to a variable would make them get rid of that process.

**Comment by andreynering** [comment]
@sascha-andres I'm OK with that. 👍 

What's you opinion on change proposed on https://github.com/go-task/task/issues/11 ?

E.g. change `vars` attribute and `Taskvars` file so it will be actually a command, and the result will be assigned to the var?

It's a backward incompatible change, but we'd be doing that early. If we decide to follow that, for consistency maybe that should be done on `env` props, too?

**Comment by sascha-andres** [comment]
@andreynering I would keep env properties as they are - you can fill in variables there. I think that would complicate stuff.

Regarding vars we could go for a fallback scenario like trying to execute and when it fails assign the text. One thing that comes into my mind is, that even for a short Taskfile the number of processes spawned might be very high.

**Pull request merged**